SYCL/ESIMD/api/functional/functions/functions_select_2d.hpp

```diff
@@ -0,0 +1,286 @@
+//===-- functions_select_2d.hpp - Functions for tests on simd rvalue select
+//      function. ---------------------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+///
+/// \file
+/// This file provides common code for tests on simd 2d select function.
+///
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#include "../shared_element.hpp"
+#include "common.hpp"
+
+namespace esimd = sycl::ext::intel::esimd;
+
+namespace esimd_test::api::functional::functions {
+
+// Descriptor class for the case of calling simd<T,N>::select function.
+struct select_2d {
+  static std::string get_description() { return "2d select"; }
+
+  template <typename DataT, int NumElems, int SizeY, int StrideY, int SizeX,
+            int StrideX, int Height, int Width>
+  static void call_esimd_function(const DataT *const ref_1,
+                                  const DataT *const ref_2, DataT *const out,
```

Comment:
OK, here we go with comment from https://github.com/intel/llvm-test-suite/pull/875#discussion_r824747657 again

Reply:
Naming and other cases has been fixed in this commit: https://github.com/intel/llvm-test-suite/pull/908/commits/ce54fc2c173d7fdf6944b2e91334eba4dbf7f90c

Reply:
What's the point in changing `ref_1` and `ref_2` into `initial_data_1` and `initial_data_2`?

You already have `initial_data` and `data_for_change` from your commit in `[functions_1d_select.hpp](https://github.com/intel/llvm-test-suite/pull/875/files#diff-295141c99d991cbb4971abc3bd754b2f17400cade3b579ac7bc36b28a1450dc3R43)`

Looks like that's the level of violating [llvm coding standards](https://llvm.org/docs/CodingStandards.html#name-types-functions-variables-and-enumerators-properly) we might easily avoid at no cost.